### DISCUSSION TITLE
Debug additional crates?

### DISCUSSION THREAD

**Discussion by hds** [SeaQL/FireDBG.for.Rust#39] [Q&A] [answered]
Is it possible to add specific crates to what FireDBG analyzes?

My use case is that I'm trying to track down an error, but due to the structure of the project, the code I need to analyze is distributed across 3 crates in the same workspace, and is invoked by an integration test in one of those crates.

Currently, I only see the code in the integration test, which isn't sufficient for my needs.

I'd just like to add that FireDBG is incredibly easy to set up. It took me no time at all to get running on the project I'm working on and that's amazing!

**Comment by tyt2y3** [comment]
Thank you for your interest. I think it will be doable if you have all crates in the same workspace. Just asking, do you have all the source code in your project directory, or are some of those crates from crates.io?

> I'd just like to add that FireDBG is incredibly easy to set up

Thank you for the positive affirmation. We did spend great deal of effort into UX engineering. I almost went bankrupt funding FireDBG. Jokes aside, I am working on async support right now, and should be able to allocate more resources after SeaORM 1.0

**Reply by hds** [reply]
All the crates are in the same workspace with the test itself. Is there somewhere I should be looking for where to enable instrumenting these crates?

**Reply by billy1624** [reply, marked as answer]
Hey @hds, by default FireDBG will only instrument the crate where the execute live in. If you want to instrument other crates in the same workspace, you need to create a `firedbg.toml` to configure it. 

https://firedbg.sea-ql.org/blog/#firedbgtoml-config

```toml
# firedbg.toml

[workspace.members]
quicksort = { trace = "full" } 
# Syntax: <PACKAGE> = { trace = "<full | none>" }
```

**Reply by hds** [reply]
Thanks for the suggestion. I tried this, but ended up having trouble with the executable crashing when I included full traces for additional packages. I'll try to dig into why, but this answer is what I needed!

**Reply by billy1624** [reply]
Hey @hds, let me know if you need help :)
